src/query_builder.rs

```diff
@@ -0,0 +1,233 @@
+            };
+
+            self.args
+                .iter()
+                .enumerate()
+                .map(generate_var_names)
+                .collect::<Vec<_>>()
+                .join(",")
+        };
+        format!(
+            "
+            CREATE {force} AGGREGATE {name}({args})
+            RETURNS {return_type}
+            WITH STATE {state}
+            AS WASM ABI {abi} 
+            {wasm_source}
+            {wit_source}
+            INITIALIZE WITH {init}
+            ITERATE WITH {iter}
+            MERGE WITH {merge}
+            TERMINATE WITH {terminate}
+            {serialize}
+            {deserialize}
+            ",
+            name = self.shared.name,
+            force = if self.shared.force { "OR REPLACE" } else { "" },
+            return_type = self.return_type,
+            state = self.state_type,
+            abi = self.shared.abi_type,
+            init = imports.init.expect("Missing init function"),
+            iter = imports.iter.expect("Missing iterate function"),
+            merge = imports.merge.expect("Missing merge function"),
+            terminate = imports.terminate.expect("Missing terminate function"),
+            serialize = imports
+                .serialize
+                .as_ref()
+                .map(|s| format!("SERIALIZE WITH {}", s))
+                .unwrap_or_default(),
+            deserialize = imports
+                .deserialize
+                .as_ref()
+                .map(|s| format!("DESERIALIZE WITH {}", s))
+                .unwrap_or_default(),
+        )
+    }
+}
+
+impl QueryBuilder for Functions {
+    fn build(&self, args: &mut Vec<String>) -> String {
+        match self {
+            Functions::Udf(f) => f.build(args),
+            Functions::Tvf(f) => f.build(args),
+            Functions::Udaf(f) => f.build(args),
+        }
+    }
+}
+
+/// Automatically try to extract the aggregate member functions from the WIT spec.
+/// Note: This requires the specific members functions to contain relevant keywords in their name.
+fn try_extract_member_functions_from_wit(source: String) -> HashMap<MemberFunction, String> {
+    let mut functions = HashMap::new();
+    for line in source.lines() {
+        if line.contains("func(") {
+            // Take the string up to the first colon
+            if let Some(name) = line.split(':').next() {
+                if let Some(func) = MemberFunction::from_name(name) {
+                    if let Some(duplicate) = functions.insert(func, name.replace("-", "_")) {
+                        eprintln!("Duplicate match for function name: {name} vs {duplicate}");
+                        exit(1);
+                    }
+                }
+            }
+        }
+    }
+    functions
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+enum MemberFunction {
+    Init,
+    Iterate,
+    Merge,
+    Terminate,
+    Serialize,
+    Deserialize,
+}
+
+impl MemberFunction {
+    fn from_name(name: &str) -> Option<Self> {
```

Comment:
Do we use the aliases you're checking for anywhere in the engine?  (e.g. "update" for "iter")

Curious how you determined the alias names.

Reply:
I added a selection of terms  I thought would be commonly used. The function names as provided by the C++ & rust template extensions are able to be auto-detected here.  I figured it would be another good reason to "standardize" on the format of an `extension`.

W.r.t. having these in the SingleStoreDB engine, I thought that might be too risky. Maybe we should hold off on that until we get to a proper structure for the `extensions` :smile: